src/main/java/capone/commands/DeadlineCommand.java

```diff
@@ -0,0 +1,114 @@
+package capone.commands;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.ArrayList;
+
+import capone.Parser;
+import capone.Storage;
+import capone.TaskList;
+import capone.Ui;
+import capone.exceptions.CaponeException;
+import capone.exceptions.InsufficientArgumentException;
+import capone.tasks.Deadline;
+
+/**
+ * Represents a command to add a deadline task to TaskList.
+ * Extends the abstract class Command.
+ *
+ * @author Tay Rui-Jie
+ */
+public class DeadlineCommand extends Command {
+
+    /** List containing input parameters for DeadlineCommand. */
+    private final ArrayList<String> inputList;
+
+    /**
+     * Constructs a DeadlineCommand with the specified input list.
+     *
+     * @param inputList The list containing input parameters for the DeadlineCommand.
+     */
+    public DeadlineCommand(ArrayList<String> inputList) {
+        this.inputList = inputList;
+    }
+
+    /**
+     * Executes the DeadlineCommand, adding a new deadline task to the TaskList.
+     *
+     * @param taskList The TaskList to be updated.
+     * @param ui       The Ui to interact with the user.
+     * @param storage  The Storage for saving data.
+     * @throws CaponeException If any Capone-related exception occurs.
+     * @inheritDoc
+     */
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws CaponeException {
+        // If the inputList has only one string, throw error (insufficient args).
+        if (inputList.size() == 1) {
+            throw new InsufficientArgumentException("Insufficient arguments!\n"
+                    + "Usage: deadline [description] /by [date]");
+        }
+
+        // Find the index of the /by command.
+        int byNdx = inputList.indexOf("/by");
+
+        // Catch potential errors from date entry.
+        if (byNdx == inputList.size() - 1 || byNdx == -1) {
+            throw new InsufficientArgumentException("Please enter a date for this deadline task!\n"
+                    + "Usage: deadline [description] /by [date]");
+        }
+
+        // Combine description of task into one string.
+        StringBuilder description = new StringBuilder();
```

Comment:
Maybe instead of using a loop, you could try using String.join() to concatenate the description parts from 1 to byNdx and String.substring() to extract the deadline part after /by

Reply:
Thank you! I will look into it 👍